frontend/ui/widget/textviewer.lua

```diff
@@ -272,6 +274,16 @@ function TextViewer:init(reinit)
         show_parent = self,
     }
 
+    -- NT: add titlebar.left_button (hamburger menu) to FocusManager.
+    if Device:hasDPad() and not (Device:hasSymKey() or Device:hasScreenKB()) then
+        -- ButtonTable calls refocusWidget on init, but we'll mangle the layout,
+        -- so kill the initial highlight while FocusManager can still find the current focused item...
+        self.button_table:handleEvent(Event:new("Unfocus"))
+        table.insert(self.button_table.layout, 1, { self.titlebar.left_button })
+        -- And refocus manually on the *actual* layout
+        self.button_table:refocusWidget()
+    end
```

Comment:
> another stray widget that needs its menu mapped to Menu key. borrowing code from @NiLuJe to fix focusmanager shenanigans.

What was the previous behaviour ?
Aren't we doing this kind of things elsewhere (or is it just focusmanager that does that?) - and if yes, why didn't we need this unfocus/refocus?
And if it is just focusmanager that does that, wouldn't it be better if it was still only it, ie. by adding/using some method?
Not really familiar with all that, so just pinging @NiLuJe to confirm this is ok.

Reply:
Remember how we once discussed always showing focus, not just after pressing a button? Well we are fixing that now, by we I mean @NiLuJe, in any case we must make sure focus is always accurate now. BTW i stole this from him ;)

Reply:
As to why we need to unfocus refocus see https://github.com/koreader/koreader/pull/12361#issuecomment-2293177294 dictionary one is exactly the same case as this.

Reply:
> Aren't we doing this kind of things elsewhere (or is it just focusmanager that does that?) - and if yes, why didn't we need this unfocus/refocus?
> And if it is just focusmanager that does that, wouldn't it be better if it was still only it, ie. by adding/using some method?
> Not really familiar with all that, so just pinging @NiLuJe to confirm this is ok.

Like the comment says, FocusManager has a method to do it (refocusWidget), and ButtonDialog calls int (on... init? for some reason, probably because a lot of stuff besides instantiation calls init I'd wager).

But the issue is that for the unfocus that happens in there to work, FocusManager has to be able to *find* the widget at the right coords in self.layout.

But here, we're completely mangling self.layout outside of FocusManager, so it completely breaks that.

Hence: manual unfocus => mangle layout => refocus

Reply:
> Hence: manual unfocus => mangle layout => refocus

Ok, fine with me if fine with you.

Reply:
Yup, see the DQL commit in #12361 (which is where that comment originated)